src/ui/public/dialogues/confirm_dialogue.js

```diff
@@ -0,0 +1,63 @@
+import uiModules from 'ui/modules';
+import confirmDialogueTemplate from './confirm_dialogue.html';
+
+const module = uiModules.get('kibana');
+
+module.factory('showConfirmDialogue', function (ModalDialogue, $q, $compile, $rootScope) {
+  /**
+   * Shows a modal confirmation dialog with the given message.
+   *
+   * @param {String} message
+   * @param {String} yesButtonText - Defaults to Okay, supply a custom string to override.
+   * @param {String} noButtonText - Defaults to Cancel, supply a custom string to override.
+   * @return {Promise<boolean>} Returns a promise that will be resolved to true if the user
+   * clicks the yes/okay button and rejected if the user clicks the no/cancel button.
+   */
+  return function showConfirmDialogue(message, yesButtonText = 'Okay', noButtonText = 'Cancel') {
```

Comment:
I disagree on this one. :)  I don't want to require callers to pass in text manually on every call site.  If we ever wanted to changed that default text, we'd have to do so everywhere.